utils/warnings.ml

```diff
@@ -471,10 +477,10 @@ let message = function
   | Expect_tailcall ->
       Printf.sprintf "expected tailcall"
   | Fragile_literal_pattern ->
-      Printf.sprintf
+      Format.asprintf
```

Comment:
I am skeptical that mixing `Printf` and `Format` is a good idea in  general. it seems safe enough here as we use the `sprintf` variants anyway, but why not stick to Printf everywhere?

Reply:
Good point, fixed.